eland/ml/exporters/es_gb_models.py

```diff
@@ -0,0 +1,449 @@
+)
+from sklearn.tree import DecisionTreeClassifier, DecisionTreeRegressor
+from sklearn.utils.validation import check_array
+
+from eland.common import ensure_es_client
+from eland.ml.common import TYPE_CLASSIFICATION, TYPE_REGRESSION
+
+from ._sklearn_deserializers import Tree
+
+
+class ESGradientBoostingModel(ABC):
+    """
+    Abstract class for converting Elastic ML model into sklearn Pipeline.
+    """
+
+    def __init__(
+        self,
+        es_client: Union[str, List[str], Tuple[str, ...], "Elasticsearch"],
+        model_id: str,
+    ) -> None:
+        """
+        Parameters
+        ----------
+        es_client : Elasticsearch client argument(s)
+            - elasticsearch-py parameters or
+            - elasticsearch-py instance
+        model_id : str
+            The unique identifier of the trained inference model in Elasticsearch.
+
+        Raises
+        ------
+        RuntimeError
+            On failure to retrieve trained model information to the specified model ID.
+        ValueError
+            The model is expected to be trained in Elastic Stack. Models initially imported
+            from xgboost, lgbm, or sklearn are not supported.
+        """
+        self.es_client: Elasticsearch = ensure_es_client(es_client)
+        self.model_id = model_id
+
+        self._trained_model_result = self.es_client.ml.get_trained_models(
+            model_id=self.model_id,
+            decompress_definition=True,
+            include=["hyperparameters", "definition"],
+        )
+
+        if (
+            "trained_model_configs" not in self._trained_model_result
+            or len(self._trained_model_result["trained_model_configs"]) == 0
+        ):
+            raise RuntimeError(
+                f"Failed to retrieve the trained model for model ID {self.model_id!r}"
+            )
+
+        if "metadata" not in self._trained_model_result["trained_model_configs"][0]:
+            raise ValueError(
+                "Error initializing sklearn classifier. Incorrect prior class probability. "
+                + "Note: only export of models trained in the Elastic Stack is supported."
+            )
+
+        preprocessors = []
+        if "preprocessors" in self._definition:
+            preprocessors = self._definition["preprocessors"]
+        self.feature_names_in_, self.input_field_names = self._get_feature_names_in_(
+            preprocessors,
+            self._definition["trained_model"]["ensemble"]["feature_names"],
+            self._trained_model_result["trained_model_configs"][0]["input"][
+                "field_names"
+            ],
+        )
+
+        feature_names_map = {name: i for i, name in enumerate(self.feature_names_in_)}
+
+        trained_models = self._definition["trained_model"]["ensemble"]["trained_models"]
+        self._trees = []
+        for trained_model in trained_models:
+            self._trees.append(Tree(trained_model["tree"], feature_names_map))
+
+        self.n_estimators = (
```

Comment:
This line got formatted weirdly, perhaps move the comment above the line and reformat?

Reply:
fixed